### ISSUE TITLE
facts must use current operation env (brew.packages latest param does not seem to be working)

### ISSUE THREAD

**Issue by taranlu-houzz** [Fizzadar/pyinfra#540]
**Describe the bug**
Using the `latest` param for `brew.packages` operation seems to just attempt a regular install (`brew install`) instead of `brew upgrade`.

**To Reproduce**
Example deploy:
```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-


"""Pyinfra deploy to debug brew latest issue.
"""


###############################################################################
# Imports
###############################################################################


# ==Site-Packages==
from pyinfra.operations import brew


###############################################################################
# Main
###############################################################################


brew.packages(
    name="Install pipx via brew.",
    packages=[
        "pipx",
    ],
    latest=True,
    update=True,
    env={"PATH": "/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"},
)

```
Example output:
```sh
❯ pyinfra inventories/----.py --limit ---- deploys/tasks/python/brew_latest_bug.py --debug -vvv
    [pyinfra_cli.main] Checking potential directory: deploys/tasks/python
    [pyinfra_cli.main] Checking potential directory: deploys/tasks
    [pyinfra_cli.main] Checking potential directory: inventories
    [pyinfra_cli.main] Checking potential directory:
    [pyinfra_cli.main] Setting directory to:
--> Loading config...
--> Loading inventory...
    [pyinfra_cli.inventory] Creating fake inventory...
    [pyinfra_cli.inventory] Looking for group data in: group_data/all.py

--> Connecting to hosts...
    [pyinfra.api.connectors.ssh] Connecting to: ---- ({'allow_agent': True, 'look_for_keys': True, 'hostname': '----', 'timeout': 10})
    [----] Connected
    [pyinfra.api.state] Activating host: ----

--> Preparing operations...
    Loading: deploys/tasks/python/brew_latest_bug.py
    [pyinfra.api.operation] Adding operation, {'Install pipx via brew.'}, opOrder=(23,), opHash=ad8996774bf8bb42ffe38b3902cd74dbe2b53114
    [pyinfra.api.facts] Getting fact: brew_packages  (ensure_hosts: (Host(----),))
    [pyinfra.api.connectors.ssh] Running command on ----: (pty=False) sh -c '! command -v brew > /dev/null || (brew list --versions)'
[----] >>> sh -c '! command -v brew > /dev/null || (brew list --versions)'
    [pyinfra.api.connectors.ssh] Waiting for exit status...
    [pyinfra.api.connectors.ssh] Command exit status: 0
    Loaded fact brew_packages
    [----] Ready: deploys/tasks/python/brew_latest_bug.py

--> Proposed changes:
    Groups: ----
    [----]   Operations: 1   Commands: 2

--> Beginning operation run...
--> Starting operation: Install pipx via brew.
    [pyinfra.api.operations] Starting operation Install pipx via brew. on ----
    [pyinfra.api.connectors.ssh] Running command on ----: (pty=None) sh -c 'env PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin brew update'
[----] >>> sh -c 'env PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin brew update'
[----] ==> Homebrew has enabled anonymous aggregate formula and cask analytics.
[----] Read the analytics documentation (and how to opt-out) here:
[----]   https://docs.brew.sh/Analytics
[----] No analytics have been recorded yet (nor will be during this `brew` run).
[----]
[----] ==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
[----]   https://github.com/Homebrew/brew#donations
[----] Updated 1 tap (homebrew/core).
[----]
[----] You have 3 outdated formulae installed.
[----] You can update them with brew upgrade.
[----] ==> Updated Formulae
[----] libomp
    [pyinfra.api.connectors.ssh] Waiting for exit status...
    [pyinfra.api.connectors.ssh] Command exit status: 0
    [pyinfra.api.connectors.ssh] Running command on ----: (pty=None) sh -c 'env PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin brew install pipx'
[----] >>> sh -c 'env PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin brew install pipx'
[----] ==> Downloading https://homebrew.bintray.com/bottles/mpdecimal-2.5.1.big_sur.bottle.tar.gz
[----] Already downloaded: /Users/3d/Library/Caches/Homebrew/downloads/29dd7202ebb6142202c80dec9080c1a24ef37a7039f92356a122061477efce21--mpdecimal-2.5.1.big_sur.bottle.tar.gz
[----] ==> Downloading https://homebrew.bintray.com/bottles/python%403.9-3.9.2_1.big_sur.bottle.tar.gz
[----] Already downloaded: /Users/3d/Library/Caches/Homebrew/downloads/812addf569674782d8f470fcc4db3abc5f0e1abb854008843ddcc383cce0b614--python@3.9-3.9.2_1.big_sur.bottle.tar.gz
[----] ==> Downloading https://homebrew.bintray.com/bottles/pipx-0.16.1.0.big_sur.bottle.tar.gz
[----] Error: pipx 0.16.0.0 is already installed
[----] To upgrade to 0.16.1.0, run:
[----]   brew upgrade pipx
[----] Already downloaded: /Users/3d/Library/Caches/Homebrew/downloads/a8ca67c1e2a7c45e3d7b13368aa1e95c93bb11f79d50e3cd0642cd7cb719d0b4--pipx-0.16.1.0.big_sur.bottle.tar.gz
    [pyinfra.api.connectors.ssh] Waiting for exit status...
    [pyinfra.api.connectors.ssh] Command exit status: 1
    [----] Error
    [pyinfra.api.state] Failing hosts: ----
--> pyinfra error: No hosts remaining!

```

**Expected behavior**
Would expect resulting command to be `brew upgrade` instead of `brew install`.

**Comment by Fizzadar** [comment]
@taranlu-houzz this looks like an issue with the brew packages fact, specifically this bit:

```
    [pyinfra.api.connectors.ssh] Running command on ----: (pty=False) sh -c '! command -v brew > /dev/null || (brew list --versions)'
[----] >>> sh -c '! command -v brew > /dev/null || (brew list --versions)'
    [pyinfra.api.connectors.ssh] Waiting for exit status...
    [pyinfra.api.connectors.ssh] Command exit status: 0
    Loaded fact brew_packages
```

There should be some output here from brew; what happens if you run `brew list --versions` manually?

**Comment by taranlu-houzz** [comment]
@Fizzadar I get the full, expected output when running `brew list --versions`:
```
❯ brew list --versions
aom 2.0.2
asdf 0.8.0
autoconf 2.69
automake 1.16.3
bash 5.1.4
bat 0.17.1
coreutils 8.32
docbook 5.1_1
docbook-xsl 1.79.2_1
exa 0.9.0_2
fd 8.2.1
fish 3.1.2
freetype 2.10.4
fzf 0.25.1
gawk 5.1.0
gdbm 1.18.1_1
gettext 0.21
ghostscript 9.53.3_1
git 2.30.1
git-lfs 2.13.2
glib 2.66.7
gmp 6.2.1
gnu-getopt 2.36.2
htop 3.0.5
ilmbase 2.5.5
imagemagick 7.0.11-1
jpeg 9d
libde265 1.0.8
libevent 2.1.12
libffi 3.3_2
libheif 1.11.0
liblqr 0.4.2_1
libomp 11.0.1
libpng 1.6.37
libtermkey 0.22
libtiff 4.2.0
libtool 2.4.6_2
libuv 1.41.0
libvterm 0.1.4
libyaml 0.2.5
little-cms2 2.12
luajit-openresty 20201229_1
mpfr 4.1.0
msgpack 3.3.0
ncurses 6.2
neofetch 7.1.0
neovim 0.4.4_1
openexr 2.5.5
openjpeg 2.4.0
openssl@1.1 1.1.1j
pcre 8.44
pcre2 10.36
pipx 0.16.0.0
python@3.9 3.9.2
readline 8.1
ripgrep 12.1.1
screenresolution 1.6
shared-mime-info 2.1
sqlite 3.34.0
starship 0.50.0
tcl-tk 8.6.11_1
tmux 3.1c_1
tree 1.8.0
unibilium 2.1.0
unixodbc 2.3.9
utf8proc 2.6.1
webp 1.2.0
x265 3.4_2
xmlto 0.0.28
xz 5.2.5
zlib 1.2.11
```

The `brew` command should be available since I am explicitly setting `PATH`. This is on a new mac mini (intel based, not M1) running the latest macOS 11.

**Comment by Fizzadar** [comment]
I believe this is because the `env` parameter isn't "carried" into the fact - so the `brew` command is missing within the fact itself. I will add this such that it is carried over.

**Comment by Fizzadar** [comment]
I've now changed the facts  API to correctly pass through the `env` param from the current operation in https://github.com/Fizzadar/pyinfra/commit/8530b3297dedd39f7a4a2ca947c1eec69a22f8fa, released  in `v1.3.8`.
